test/test_linearizer.py

```diff
@@ -433,7 +433,7 @@ def test_double_reduce(self):
     if not isinstance(Device[Device.DEFAULT], Compiled) or not Device[Device.DEFAULT].linearizer_opts.has_local or not Device[Device.DEFAULT].linearizer_opts.has_shared:
       self.skipTest("Only Compiled uses linearizer with locals and shared")
 
-    N = 128
+    N = 32  # set N such that the sum is not split into 2 kernels
```

Comment:
cc @flammit i lowered this so that the sum is not splitting into two kernels with the new heuristic. Please let me know if this breaks any assumptions you have for this test.

Reply:
I think the sizes of the shifts just need to be brought down as well (change the 32/16 down to 4/2 or something) so the structure stays the same.